src/Core/Silk.NET.BuildTools/Bind/StructWriter.cs

```diff
@@ -475,22 +490,53 @@ CallingConvention conv
             sw.Flush();
         }
 
-        public static void WriteFusedField(Field field, List<string> args, StreamWriter sw)
+        public static void WriteFusedField(Struct @struct, Project p, Field field, List<string> args, StreamWriter sw)
         {
-            sw.WriteLine("#if NETSTANDARD2_1");
-            sw.WriteLine($"        public ref {field.Type} {field.Name}");
-            sw.WriteLine("        {");
-            sw.WriteLine("            [MethodImpl((MethodImplOptions) 768)]");
-            sw.WriteLine($"            get => ref {args[1]}.{args[2]};");
-            sw.WriteLine("        }");
-            sw.WriteLine("#else");
+            var temporaryValue = IsTemporaryValue(p, @struct, args);
+            if (!temporaryValue)
+            {
+                sw.WriteLine("#if NETSTANDARD2_1 || NETCOREAPP3_1 || NET5_0 || NET5_0_OR_GREATER");
+                sw.WriteLine($"        public ref {field.Type} {field.Name}");
+                sw.WriteLine("        {");
+                sw.WriteLine("            [MethodImpl((MethodImplOptions) 768)]");
+                sw.WriteLine($"            get => ref MemoryMarshal.CreateSpan(ref {args[1]}, 1)[0].{args[2]};");
```

Comment:
Pre-emptive response to a potential review comment: the old code that was in the NS21+ branch didn't actually compile, and we never spotted it because for some reason the DirectX projects were NS20 only.